lib/concurrent-ruby/concurrent/timer_task.rb

```diff
@@ -296,10 +340,12 @@ def schedule_next_task(interval = execution_interval)
     # @!visibility private
     def execute_task(completion)
       return nil unless @running.true?
+      start = Concurrent.monotonic_time
       _success, value, reason = @executor.execute(self)
       if completion.try?
         self.value = value
-        schedule_next_task
+        interval = interval_type == FIXED_DELAY ? execution_interval : [execution_interval - (Concurrent.monotonic_time - start), 0].max
```

Comment:
This is quite a long line, how about a `if/else` or a `case/when`?

Reply:
I think it would be simpler if it was a sub-class. This PR would only be a 2-3 line change in that case.